src/camera_session.cpp

```diff
@@ -29,8 +29,19 @@ using namespace libcamera;
 
 CameraSession::CameraSession(const CameraManager* const cm) {
   PRINT_FUNC();
-  if (opts.camera < cm->cameras().size()) {
+  if (opts.camera < 0) {
+    for (size_t i = 0; i < cm->cameras().size(); ++i) {
+      camera_ = cm->cameras()[i];
+      if (!CameraSession::validateConfig())
+        return;
+    }
+  } else if (opts.camera < static_cast<int>(cm->cameras().size())) {
     camera_ = cm->cameras()[opts.camera];
+    if (CameraSession::validateConfig()) {
+      EPRINT("Camera configuration invalid\n");
```

Comment:
Please don't double log, this string would have already been printed in validateConfig